---
int32 error
string ip_address
string netmask
string gateway